modules/issue_tracker/php/endpoints/attachment.class.inc

```diff
@@ -0,0 +1,259 @@
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse POST request body.
+        $values     = $request->getQueryParams();
+        $user       =& \User::singleton();
+        $attachment = new \LORIS\issue_tracker\UploadHelper();
+        $response   = $attachment->setupUploading(
+            $user,
+            $_FILES,
+            $values
+        );
+        return new \LORIS\Http\Response\JsonResponse(
+            $response
+        );
+    }
+
+    /**
+     * Handle DELETE requests for attachment
+     * and marks attachment as deleted from issue_tracker.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleDELETE(ServerRequestInterface $request)
+    : ResponseInterface
+    {
+        // Verify User can delete file.
+        if ($this->_deletePermission($request)) {
+            // Parse request body.
+            $values    = $request->getQueryParams();
+            $file_hash = $values['file_hash'];
+            $DB        = \Database::singleton();
+            $query     = 'UPDATE
+                             issues_attachments
+                          SET
+                             deleted = 1
+                          WHERE
+                             file_hash = :file_hash;';
+            $stmt      = $DB->prepare($query);
+            $stmt->execute(array('file_hash' => $file_hash));
+
+            return new \LORIS\Http\Response\JsonResponse(
+                array('success' => true)
+            );
+        }
+        return new \LORIS\Http\Response\JsonResponse(
+            array('error' => 'invalid permission')
+        );
+    }
+
+    /**
+     * Check if user is able to delete the attachment
+     * from the issue_tracker module.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return bool The permission to delete attachment.
+     */
+    private function _deletePermission(ServerRequestInterface $request)
+    : bool
+    {
+        // Parse request body.
+        $values   = $request->getQueryParams();
+        $user     =& \User::singleton();
+        $username = $user->getUsername();
+        $ID       = $values['ID'];
+        $DB       = \Database::singleton();
+        $query    = 'SELECT
+                        USER
+                    FROM
+                        issues_attachments
+                    WHERE
+                        ID = :ID
+                        AND USER = :user;';
+        $result   = $DB->pselect(
+            $query,
+            array('ID' => $ID, 'user' => $username)
+        );
+        // Check if User shouldn't be able to Delete.
+        if (empty($result)
+            && !$user->hasAnyPermission(
```

Comment:
If you're just checking one permission, use `hasPermission()` instead

Reply:
@johnsaigle updated in latest commit. I do think it would be nice to just have one function that does either check.